### PR TITLE
Swagger PR for new controller endpoint in MarketplaceRP

### PR DESCRIPTION
**Pull request by omshalev**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by raych1** [comment]
@omshalev , please fix the CI failures.

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2019-12-01/Marketplace.json [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    "https"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Marketplace/privateStoreClient/isPrivateClient": {
+      "get": {
+        "tags": [
+          "isPrivateClient"
+        ],
+        "description": "Check if client is private or not.",
+        "operationId": "GetPrivateStoreClient",
```
Comment (raych1) [inline specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2019-12-01/Marketplace.json]: Please keep operationId with consistent format: `Resource_Action`.

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2019-12-01/Marketplace.json [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded."
+          },
+          "404": {
```
Comment (raych1) [inline specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2019-12-01/Marketplace.json]: No need to model errors. `Default` handles all errors.

**Comment by raych1** [comment]
@MyronFanQiu, @ArcturusZhang, @qiaozha , @erich-wang, @zikalino, please help review the SDK readme files since this is a new RP.

**Comment by qiaozha** [comment]
@omshalev  please change the pakage-name, output-folder of readme.typescript.md like https://raw.githubusercontent.com/omshalev/azure-rest-api-specs/master/specification/storage/resource-manager/readme.typescript.md this format. So that it is consistent with SDK.

**Comment by omshalev** [comment]
@raych1 and @qiaozha can you please recheck and approve? Sorry for pushing but this PR is part of JEDI project and its blocking us from deployment. Thanks !

**Comment by qiaozha** [comment]
> @raych1 and @qiaozha can you please recheck and approve? Sorry for pushing but this PR is part of JEDI project and its blocking us from deployment. Thanks !

your package-name should be like `@azure/arm-marketplace`.
And also I would recommend you to add a  `clear-output-folder: true` option in readme.typescript.md. because it might cause some reference problem while generating typescript SDK code when you try to delete some definitions of swagger in the future if you don't add this option.

**Comment by omshalev** [comment]
> > @raych1 and @qiaozha can you please recheck and approve? Sorry for pushing but this PR is part of JEDI project and its blocking us from deployment. Thanks !
> 
> your package-name should be like `@azure/arm-marketplace`.
> And also I would recommend you to add a `clear-output-folder: true` option in readme.typescript.md. because it might cause some reference problem while generating typescript SDK code when you try to delete some definitions of swagger in the future if you don't add this option.

Done

**Comment by qiaozha** [comment]
okay for typescript SDK now @raych1

**Comment by omshalev** [comment]
@raych1 can we move forward?
Thanks

**Comment by raych1** [comment]
@omshalev , it looks good to me. But, do you confirm it works for all the other SDK generation?

**Comment by omshalev** [comment]
> @omshalev , it looks good to me. But, do you confirm it works for all the other SDK generation?

It should work, these SDKs are just for generating the swagger documentation using other languages right? Is this something we should take care of on our side?

**Comment by omshalev** [comment]
> > @omshalev , it looks good to me. But, do you confirm it works for all the other SDK generation?
> 
> It should work, these SDKs are just for generating the swagger documentation using other languages right? Is this something we should take care of on our side?

I just follwed an old PR someone did on marketplace here

**Comment by omshalev** [comment]
> @omshalev , it looks good to me. But, do you confirm it works for all the other SDK generation?

Please answer, we really need this to be signedoff

File: specification/marketplace/resource-manager/readme.go.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  clear-output-folder: true
+```
+
+### Tag: package-2019-12-01 and go
+
+These settings apply only when `--tag=package-2019-12-01 --go` is specified on the command line.
+Please also specify `--go-sdks-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+```yaml $(tag) == 'package-2019-12-01' && $(go)
+namespace: Microsoft.Marketplace
```
Comment (ArcturusZhang) [inline specification/marketplace/resource-manager/readme.go.md]: Please change the namespace to `marketplace`

File: specification/marketplace/resource-manager/readme.go.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+output-folder: $(go-sdks-folder)/preview/marketplace/Generated
```
Comment (ArcturusZhang) [inline specification/marketplace/resource-manager/readme.go.md]: Please change the `output-folder` to 
```
output-folder: $(go-sdk-folder)/services/marketplace/mgmt/2019-12-01/$(namespace)
```

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/marketplace/resource-manager/readme.go.md (hunk 3 above)
Comment (omshalev) [reply specification/marketplace/resource-manager/readme.go.md]: no need for SDK, removed them all

File: specification/marketplace/resource-manager/readme.go.md (hunk 4 above)
Comment (omshalev) [reply specification/marketplace/resource-manager/readme.go.md]: no need for SDK, removed them all

**Review by ArcturusZhang — APPROVED** [review]

**Review by raych1 — APPROVED** [review]

**Pull request merged**